tests/functional/specs/image_segmentation/tools/selection-tool.cy.ts

```diff
@@ -0,0 +1,346 @@
+        .data(simpleImageData)
+        .withResult(fourRectanglesResult)
+        .init();
+
+      ImageView.waitForImage();
+
+      ImageView.selectMoveToolByButton();
+      ImageView.drawRectRelative(.1, .1, .8, .8);
+      Sidebar.hasSelectedRegions(4);
+
+      ImageView.clickAtRelative(.1, .1, { ctrlKey: true, metaKey: true });
+      Sidebar.hasSelectedRegions(4);
+    });
+  });
+  describe('Selecting area', () => {
+    it('Should be able to select just one region', () => {
+      LabelStudio.params()
+        .config(simpleRectangleConfig)
+        .data(simpleImageData)
+        .withResult(fourRectanglesResult)
+        .init();
+
+      ImageView.waitForImage();
+      ImageView.selectMoveToolByButton();
+      ImageView.drawRectRelative(.1, .1, .4, .4);
+      Sidebar.hasSelectedRegions(1);
+    });
+
+    it('Should be able to select all regions', () => {
+      LabelStudio.params()
+        .config(simpleRectangleConfig)
+        .data(simpleImageData)
+        .withResult(fourRectanglesResult)
+        .init();
+
+      ImageView.waitForImage();
+      ImageView.selectMoveToolByButton();
+      ImageView.drawRectRelative(.1, .1, .8, .8);
+      Sidebar.hasSelectedRegions(4);
+    });
+
+    it('Should not select hidden region by selecting area', () => {
+      LabelStudio.params()
+        .config(simpleRectangleConfig)
+        .data(simpleImageData)
+        .withResult(simpleRectangleResult)
+        .init();
+
+      ImageView.waitForImage();
+      ImageView.selectMoveToolByButton();
+      Sidebar.toggleRegionVisibility(0);
+      Sidebar.hasSelectedRegions(0);
+      ImageView.drawRectRelative(.1, .1, .8, .8);
+      Sidebar.hasSelectedRegions(0);
+    });
+
+    it('Should disappear after mouseup', () =>{
+      LabelStudio.params()
+        .config(simpleRectangleConfig)
+        .data(simpleImageData)
+        .withResult([])
+        .init();
+
+      ImageView.waitForImage();
+      ImageView.selectMoveToolByButton();
+      ImageView.capture('canvas');
+      ImageView.drawRectRelative(.05, .05, .9, .9);
+      // empirically chosen threshold to catch slight changes
+      ImageView.canvasShouldNotChange('canvas', 0.009);
+    });
+
+    it('Should add regions to selection with Ctrl pressed', () =>{
+      LabelStudio.params()
+        .config(simpleRectangleConfig)
+        .data(simpleImageData)
+        .withResult(fourRectanglesResult)
+        .init();
+
+      ImageView.waitForImage();
+      ImageView.selectMoveToolByButton();
+      Sidebar.hasSelectedRegions(0);
+
+      ImageView.drawRectRelative(.1, .1, .4, .4, { ctrlKey: true, metaKey: true });
+      Sidebar.hasSelectedRegions(1);
+
+      ImageView.drawRectRelative(.1, .1, .8, .4, { ctrlKey: true, metaKey: true });
+      Sidebar.hasSelectedRegions(2);
+
+      ImageView.drawRectRelative(.1, .5, .8, .4, { ctrlKey: true, metaKey: true });
+      Sidebar.hasSelectedRegions(4);
+
+    });
+
+    it('Should not reset selection with Ctrl pressed', () =>{
```

Comment:
also it would be great to check for regression that if we select group of regions and click on one of them _without_ Ctrl pressed — just this region will be selected even if it was part of the group.